demos/simple-gotracing/http_trace/bpf_program.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+const bpfProgram = `
+#include <uapi/linux/ptrace.h>
+/***********************************************************
+ * BPF program that traces a GO HTTP func (experimental).
+ *
+ * This program traces GO's HTTP library and extracts out
+ * internal variables: method, uri, response code, data.
+ **********************************************************/
+
+BPF_PERF_OUTPUT(golang_http_response_events);
+
+#define MSG_BUF_SIZE 128
+#define URI_BUF_SIZE 64
+#define METHOD_BUF_SIZE 8
+
+struct golang_http_response_event_t {
+  u64 status_code;
+
+  u64 uri_len;
+  char uri[URI_BUF_SIZE];
+
+  u64 method_len;
+  char method[METHOD_BUF_SIZE];
+
+  u64 msg_len;
+  char msg[MSG_BUF_SIZE];
+};
+
+int probe_golang_http_response(struct pt_regs *ctx) {
+    struct golang_http_response_event_t event = {};
+
+    // Positions within stack frame:
+    u64 struct_response_pos = 5;
+
+    // Positions within struct response object:
+    u64 req_ptr_pos = 1;
+    u64 bufio_ptr_pos = 6;
+    u64 content_length_pos = 13;
+    u64 status_code_pos = 14;
+
+    // Positions within struct request object:
+    u64 method_ptr_pos = 0;
+    u64 method_len_pos = 1;
+    u64 uri_ptr_pos = 24;
+    u64 uri_len_pos = 25;
+
+    // Positions within bufio.Writer object:
+    u64 buf_ptr_pos = 2;
+    u64 array_ptr_pos = buf_ptr_pos + 0;
+    u64 array_len_pos = buf_ptr_pos + 1;
+
+    //-------
```

Comment:
Not sure the intention of `//---------`

If for separating different "sections" of code, then include description of the purpose of each chunk of code could be useful.